### ISSUE TITLE
Document how to completely unregister Shell Namespace

### ISSUE THREAD

**Issue by AliveDevil** [MicrosoftDocs/winrt-api#1130]
After calling `StorageProviderSyncRootManager.Register()` there are registry-keys generated at several places which provide access to the sync root through the Windows Explorer (Shell Namespace).

These registry keys are not deleted after calling `StorageProviderSyncRootManager.Unregister()`.
Please document this non-cleaning behavior and eventually upgrade Windows Runtime to automatically cleanup remaining registry keys.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0daec2e2-28ee-cc36-d4a8-584cd0d77ac4
* Version Independent ID: 43760451-a28a-5f70-59f5-c4d7edf754b0
* Content: [StorageProviderSyncRootManager.Unregister(String) Method (Windows.Storage.Provider) - Windows UWP applications](https://docs.microsoft.com/en-us/uwp/api/windows.storage.provider.storageprovidersyncrootmanager.unregister#Windows_Storage_Provider_StorageProviderSyncRootManager_Unregister_System_String_)
* Content Source: [https://cpubwin.visualstudio.com/DefaultCollection/windows-uwp/_git/winrt-api-build?path=%2fwinrt-api-build%2fwindows.storage.provider.storageprovidersyncrootmanager.yml&version=GBlive&_a=contents](https://cpubwin.visualstudio.com/DefaultCollection/windows-uwp/_git/winrt-api-build?path=%2fwinrt-api-build%2fwindows.storage.provider.storageprovidersyncrootmanager.yml&version=GBlive&_a=contents)
* Product: **uwp**
* Technology: **winrt-reference**
* Git Login: **knicholasa**
* Microsoft Alias: **nichola**

**Comment by AliveDevil** [comment]
Contributing additional information here:
All remaining data for `StorageProviderSyncRootManager` is automatically removed from the system

**If** and **Only If** the application package has been installed through MSIX/APPX/Deployment. After removing the package.
It would be great if these steps would be documented for classic Win32-Apps using the Windows Runtime outside of the sandbox.

**Comment by knicholasa** [comment]
Hi @AliveDevil, thanks for leaving this feedback. Would it be possible for you to provide the keys that were not deleted so the team can investigate this further?

**Comment by knicholasa** [comment]
I've added a note about the need to unregister non-MSIX-packaged desktop apps. It should show up on the page shortly.

**Comment by AliveDevil** [comment]
I did some more tests with following result:

There is a NameSpace registered with a random UUID in `HKCU\Software\Microsoft\Windows\CurrentVersion\Explorer\Desktop\NameSpace` which is **not** cleaned after uninstalling an app, if `StorageProviderSyncRootManager` errors out while calling `Register`.

There is a key created in `HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\SyncRootManager` with the ID of the SyncRoot through `StorageProviderSyncRootManager` that is **not** cleaned once `Register` fails and I uninstall the packaged app. There is a value called `NamespaceCLSID` which corresponds to previously mentioned entry in `HKCU\..\NameSpace`.

After uninstalling the packaged app with an error while running `StorageProviderSyncRootManager.Register()` there is are orphan keys created in `HKCR\CLSID`, `HKCU\Software\Classes\CLSID` with the `NamespaceCLSID` - which is again **not** cleaned after uninstalling the packaged app.

Is there any way to force and verify that uninstalling a packaged app will not leave behind any orphan entries inside the registry rendering the Windows Explorer unusable due to having thousands of namespaces registered which are there because `StorageProviderSyncRootManager.Register` threw an exception.

tl;dr: Uninstalling a packaged app which has just tried to register a Sync Root, which failed creates orphan entries at following registry paths, which are never removed and - in case of the `HKCU\..\NameSpace` result in malfunctioning links to not existing paths because the app has just been deleted.
* `HKCU\Software\Microsoft\Windows\CurrentVersion\Explorer\Desktop\NameSpace\<NamespaceCLSID>`
* `HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\SyncRootManager\<Registered ID>`\ with reference to `NamespaceCLSID`
* `HKCR\CLSID\<NamespaceCLSID>`
* `HKCU\Software\Classes\CLSID\<NamespaceCLSID>`
